common/debug-interface/src/libra_trace.rs

```diff
@@ -296,3 +341,70 @@ fn abbreviate_crate(name: &str) -> &str {
         _ => name,
     }
 }
+
+// This is exact copy of similar function in log crate
+/// Sets libra trace config
+pub fn set_libra_trace(config: &HashMap<String, String>) -> Result<()> {
+    match parse_sampling_config(config) {
+        Ok(sampling) => unsafe {
+            match LIBRA_TRACE_STATE.compare_and_swap(UNINITIALIZED, INITIALIZING, Ordering::SeqCst)
+            {
+                UNINITIALIZED => {
+                    SAMPLING_CONFIG = Some(sampling);
+                    LIBRA_TRACE_STATE.store(INITIALIZED, Ordering::SeqCst);
+                    Ok(())
+                }
+                INITIALIZING => {
+                    while LIBRA_TRACE_STATE.load(Ordering::SeqCst) == INITIALIZING {}
+                    bail!("Failed to initialize LIBRA_TRACE_STATE");
+                }
+                _ => bail!("Failed to initialize LIBRA_TRACE_STATE"),
+            }
+        },
+        Err(s) => bail!("Failed to parse sampling config: {}", s),
+    }
+}
+
+fn parse_sampling_config(config: &HashMap<String, String>) -> Result<Sampling> {
+    let mut map = HashMap::new();
+    for (category, rate) in config {
+        let k: &'static str = Box::leak(category.clone().into_boxed_str());
+        let v = rate.split('/').collect::<Vec<&str>>();
+        ensure!(
+            v.len() == 2,
+            "Failed to parse {:?} in nominator/denominator format",
+            rate
+        );
+        let v = CategorySampling {
+            nominator: v[0].parse::<u64>()?,
+            denominator: v[1].parse::<u64>()?,
+        };
+        map.insert(k, v);
+    }
+    Ok(Sampling(map))
+}
+
+/// Checks if libra trace is enabled
+pub fn libra_trace_set() -> bool {
+    LIBRA_TRACE_STATE.load(Ordering::SeqCst) == INITIALIZED
+}
+
+pub fn is_selected(node: (&'static str, u64)) -> bool {
+    if libra_trace_set() {
```

Comment:
nit: for situations when you have early termination condition, best practice is to do something like
```
if !trace_set() {
return false
}
...do work....
```

This allows to reduce big code blocks that are hard to read